src/algebra/order/to_interval_mod.lean

```diff
@@ -457,6 +458,106 @@ begin
+    rw [neg_smul, ←sub_eq_add_neg, ne.def, eq_sub_iff_add_eq] at h,
+    apply eq_to_Ioc_div_of_add_zsmul_mem_Ioc,
+    have := add_to_Ico_div_zsmul_mem_Ico a hb x,
+    refine ⟨lt_of_le_of_ne' this.1 h, this.2.le⟩ },
+end
+
+/-- `to_Ico_mod` agrees with `to_Ioc_mod` if `x` does not lie on a boundary. -/
+lemma to_Ico_mod_eq_to_Ioc_mod (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ico_mod a hb x = to_Ioc_mod a hb x ↔ ∀ z : ℤ, x ≠ a + z • b :=
+by rw [to_Ico_mod, to_Ioc_mod, add_left_cancel_iff, (zsmul_strict_mono_left hb).injective.eq_iff,
+    to_Ico_div_eq_to_Ioc_div]
+
+/-- `to_Ico_div` disagrees with `to_Ioc_div` if `x` lies on a boundary. -/
+lemma to_Ico_div_eq_to_Ioc_div_sub_one (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ico_div a hb x = to_Ioc_div a hb x - 1 ↔ ∃ z : ℤ, x = a + z • b :=
+begin
+  split,
+  { intro h,
+    refine ⟨1 - to_Ioc_div a hb x, _⟩,
+    rw [sub_smul, one_smul, ←add_sub_assoc, eq_sub_iff_add_eq],
+    have hco := (add_to_Ico_div_zsmul_mem_Ico a hb x).1,
+    have hoc := (add_to_Ioc_div_zsmul_mem_Ioc a hb x).2,
+    rw [h, sub_smul, one_smul, ←add_sub_assoc, le_sub_iff_add_le] at hco,
+    exact le_antisymm hoc hco },
+  { rintros ⟨z, rfl⟩,
+    rw [to_Ico_div_add_zsmul, to_Ioc_div_add_zsmul, to_Ioc_div_apply_left,
+      to_Ico_div_apply_left, sub_sub_cancel_left, zero_sub] }
+end
+
+/-- `to_Ico_mod` disagrees with `to_Ioc_mod` if `x` lies on a boundary. -/
+lemma to_Ico_mod_eq_to_Ioc_mod_sub (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ico_mod a hb x = to_Ioc_mod a hb x - b ↔ ∃ z : ℤ, x = a + z • b :=
+by rw [to_Ico_mod, to_Ioc_mod, add_sub_assoc, add_left_cancel_iff, sub_eq_add_neg, ←zsmul_sub_one,
+  (zsmul_strict_mono_left hb).injective.eq_iff, to_Ico_div_eq_to_Ioc_div_sub_one]
+
+/-- `to_Ico_div` disagrees with `to_Ioc_div` if `x` lies on a boundary. -/
+lemma to_Ico_div_add_one_eq_to_Ioc_div (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ico_div a hb x + 1 = to_Ioc_div a hb x ↔ ∃ z : ℤ, x = a + z • b :=
+by rw [←to_Ico_div_eq_to_Ioc_div_sub_one, eq_sub_iff_add_eq]
+
+/-- `to_Ico_mod` disagrees with `to_Ioc_mod` if `x` lies on a boundary. -/
+lemma to_Ico_mod_add_eq_to_Ioc_mod (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ico_mod a hb x + b = to_Ioc_mod a hb x ↔ ∃ z : ℤ, x = a + z • b :=
+by rw [←to_Ico_mod_eq_to_Ioc_mod_sub, eq_sub_iff_add_eq]
+
+lemma to_Ico_div_wcovby_to_Ioc_div (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ico_div a hb x ⩿ to_Ioc_div a hb x :=
+begin
+  suffices : to_Ico_div a hb x = to_Ioc_div a hb x ∨ to_Ico_div a hb x + 1 = to_Ioc_div a hb x,
+  { obtain h | h := this,
+    { rw h,
+      exact wcovby.refl _ },
+    { rw ←h,
+      exact (int.covby_add_one _).wcovby } },
+  rw [to_Ico_div_add_one_eq_to_Ioc_div, to_Ico_div_eq_to_Ioc_div, ←not_exists],
+  exact (em _).symm
+end
+
+lemma to_Ioc_div_le_to_Ico_div_add_one (a : α) {b : α} (hb : 0 < b) (x : α) :
+  to_Ioc_div a hb x ≤ to_Ico_div a hb x + 1 :=
+begin
+  cases wcovby_iff_covby_or_eq.mp (to_Ico_div_wcovby_to_Ioc_div a hb x),
```

Comment:
@YaelDillies: is there a lemma about `order.succ` that makes this a trivial consequence of the lemma above?

Reply:
[#17940](https://github.com/leanprover-community/mathlib/pull/17940)

Reply:
I've removed this lemma now.